### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by RobWitteman** [flutter-stripe/flutter_stripe#602] [Q&A]
Hi! 

Is there a tutorial on how to add a return to merchant URL for IOS? If a make a payment with IOS, a button "return to merchant" appears.  The button shows a error message "does not recognize the url". 

Thank you in advance. 

Kind regards,
Rob

**Comment by remonh87** [comment]
hi @RobWitteman, 

you can just follow the standard flutter instructions for deep linking: https://docs.flutter.dev/development/ui/navigation/deep-linking. After that you can add the url scheme as setting. Like: https://github.com/flutter-stripe/flutter_stripe/blob/b6bb2ec72b3b45092eddd6053893e56d139de2b7/example/lib/main.dart#L11

**Reply by punxti** [reply]
Hi @RobWitteman Just wondering if you have found a solution on this already? I have got the same issue here.

@remonh87 I have the Stripe.urlScheme = 'flutterstripe'; setting. My payment flow requires server to create payment intent, and client side to confirm payment. With this flow, stripe doesn't allow me to provide return_url when creating payment intent on the server. So return_url has to be provided from client side I assume?

My question is where I can set the return url with the flutter-stripe package on client side?  Stripe.instance.confirmPayment doesn't seem to have such option. 

`final paymentIntent = await Stripe.instance.confirmPayment(
        paymentIntentClientSecret: payment.clientSecret,
        data: PaymentMethodParams.card(
          paymentMethodData: PaymentMethodData(
            billingDetails: widget.billingDetails,
            shippingDetails: widget.shippingDetails,
          ),
        ),
        options: PaymentMethodOptions(
          setupFutureUsage: widget.saveCard == true
              ? PaymentIntentsFutureUsage.OffSession
              : null,
        ),
      );`



Without providing return url, the payment works fine on android, but on iOS, I always get an extra screen with "return to merchant" button. 

Alternatively, Is there a way not to show this screen on iOS?

Thank you!
